test/core/ExamplesUnitTests/TribitsExampleProject_Tests.cmake

```diff
@@ -66,369 +66,419 @@ set(LabelsForSubprojects_REGEX
-      -DEXPORT_VAR_PREFIX=TribitsExProj
-      -DEXPORT_CONFIG_FILE=../install/lib/cmake/TribitsExProj/TribitsExProjConfig.cmake
-      -DCMAKE_COMMAND=${CMAKE_COMMAND}
-      -P ${CMAKE_CURRENT_SOURCE_DIR}/RunDummyPackageClientBulid.cmake
-    PASS_REGULAR_EXPRESSION_ALL
-      "DUMMY_PROJECT_NAME = 'DummyProject'"
-      "DUMMY_PROJECT_DIR = 'dummy_client_of_TribitsExProj'"
-      "EXPORT_CONFIG_FILE = '../install/lib/cmake/TribitsExProj/TribitsExProjConfig.cmake'"
-      "EXPORT_VAR_PREFIX = 'TribitsExProj'"
-      "CMAKE_COMMAND = '${CMAKE_COMMAND}"
-      "Create the dummy client directory ..."
-      "Create dummy dummy_client_of_TribitsExProj/CMakeLists.txt file ..."
-      "Configure the dummy project to print the variables in .*/TriBITS_TribitsExampleProject_ALL_ST_NoFortran/dummy_client_of_TribitsExProj ..."
-      "DUMMY_PROJECT_NAME = 'DummyProject'"
-      "EXPORT_CONFIG_FILE = '../install/lib/cmake/TribitsExProj/TribitsExProjConfig.cmake'"
-      "EXPORT_VAR_PREFIX = 'TribitsExProj'"
-      "Including file '.*/TriBITS_TribitsExampleProject_ALL_ST_NoFortran/dummy_client_of_TribitsExProj/../install/lib/cmake/TribitsExProj/TribitsExProjConfig.cmake'"
-      "TribitsExProj_CMAKE_BUILD_TYPE = 'RELEASE'"
-      "TribitsExProj_CXX_COMPILER = '${CMAKE_CXX_COMPILER_FOR_REGEX}'"
-      "TribitsExProj_C_COMPILER = '${CMAKE_C_COMPILER_FOR_REGEX}'"
-      "TribitsExProj_Fortran_COMPILER = ''"
-      "TribitsExProj_FORTRAN_COMPILER = ''"
-      "TribitsExProj_CXX_FLAGS = ''"
-      "TribitsExProj_C_FLAGS = ''"
-      "TribitsExProj_Fortran_FLAGS = ''"
-      "TribitsExProj_EXTRA_LD_FLAGS = ''"
-      "TribitsExProj_SHARED_LIB_RPATH_COMMAND = ''"
-      "TribitsExProj_BUILD_SHARED_LIBS = 'FALSE'"
-      "TribitsExProj_LINKER = '.*'"
-      "TribitsExProj_AR = '.*'"
-      "TribitsExProj_INSTALL_DIR = '.*/TriBITS_TribitsExampleProject_ALL_ST_NoFortran/install'"
-      "TribitsExProj_INCLUDE_DIRS = ''"
-      "TribitsExProj_LIBRARY_DIRS = ''"
-      "TribitsExProj_LIBRARIES = 'WithSubpackagesC::pws_c[;]WithSubpackagesB::pws_b[;]WithSubpackagesA::pws_a[;]SimpleCxx::simplecxx'"
-      "TribitsExProj_TPL_INCLUDE_DIRS = ''"
-      "TribitsExProj_TPL_LIBRARY_DIRS = ''"
-      "TribitsExProj_TPL_LIBRARIES = 'HeaderOnlyTpl::all_libs'"
-      "TribitsExProj_MPI_LIBRARIES = ''"
-      "TribitsExProj_MPI_LIBRARY_DIRS = ''"
-      "TribitsExProj_MPI_INCLUDE_DIRS = ''"
-      "TribitsExProj_MPI_EXEC = '.*'"
-      "TribitsExProj_MPI_EXEC_MAX_NUMPROCS = '[1-9]*'"  # Is null for an MPI build
-      "TribitsExProj_MPI_EXEC_NUMPROCS_FLAG = '.*'"
-      "TribitsExProj_PACKAGE_LIST = 'WithSubpackages[;]WithSubpackagesC[;]WithSubpackagesB[;]WithSubpackagesA[;]SimpleCxx'"
-      "TribitsExProj_TPL_LIST = 'HeaderOnlyTpl"  # Must work for no MPI too
-      "-- Configuring done"
-      "-- Generating done"
-    ALWAYS_FAIL_ON_NONZERO_RETURN
+  if (sharedOrStatic STREQUAL "SHARED")
+    set(BUILD_SHARED_LIBS_VAL ON)
+    set(libExtRegex "[.]so[.].*")
```

Comment:
Doesn't this regex *require* a `.so` file to have a suffix? There are some `.so` files that don't. Is this the desired behavior?

Reply:
TribitsExampleProject used in these tests uses SOVERSION, so they should be guaranteed to have the version suffixes.